### PR TITLE
[Merged by Bors] - feat: Data.Finset.NatAntidiagonal.antidiagonal_equiv_fin

### PR DESCRIPTION
**Pull request by nomeata**
as suggested by @ocfnash in https://github.com/leanprover-community/mathlib4/pull/6425#discussion_r1304089614

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Finset/NatAntidiagonal.lean [hunk 1]
```diff
@@ -101,18 +101,29 @@ theorem antidiagonal_congr {n : ℕ} {p q : ℕ × ℕ} (hp : p ∈ antidiagonal
...
+@[ext] theorem antidiagonal_subtype_ext {n : ℕ} {p q : antidiagonal n} (h : p.val.1 = q.val.1) :
+    p = q := Subtype.ext ((antidiagonal_congr p.prop q.prop).mpr h)
+
 theorem antidiagonal.fst_le {n : ℕ} {kl : ℕ × ℕ} (hlk : kl ∈ antidiagonal n) : kl.1 ≤ n := by
   rw [le_iff_exists_add]
   use kl.2
   rwa [mem_antidiagonal, eq_comm] at hlk
 #align finset.nat.antidiagonal.fst_le Finset.Nat.antidiagonal.fst_le
 
+theorem antidiagonal.fst_lt {n : ℕ} {kl : ℕ × ℕ} (hlk : kl ∈ antidiagonal n) : kl.1 < (n + 1) :=
```
Comment (alexjbest) [inline Mathlib/Data/Finset/NatAntidiagonal.lean:115]:
```suggestion
theorem antidiagonal.fst_lt {n : ℕ} {kl : ℕ × ℕ} (hlk : kl ∈ antidiagonal n) : kl.1 < n + 1 :=
```

**Comment by ocfnash** [comment]
Thanks!

bors d+

File: Mathlib/Data/Finset/NatAntidiagonal.lean [hunk 2]
```diff
@@ -203,6 +214,15 @@ def sigmaAntidiagonalEquivProd : (Σn : ℕ, antidiagonal n) ≃ ℕ × ℕ wher
 
 end EquivProd
 
+/-- The set `antidiagonal n` is equivalent to `Fin (n+1)`, via the first projection. --/
+def antidiagonal_equiv_fin (n : ℕ) : antidiagonal n ≃ Fin (n + 1) where
```
Comment (eric-wieser) [inline Mathlib/Data/Finset/NatAntidiagonal.lean:218]: Ideally this would have `@[simp]`

Comment (eric-wieser) [reply Mathlib/Data/Finset/NatAntidiagonal.lean:218]: Also, mathlib4 naming convention:
```suggestion
def antidiagonalEquivFin (n : ℕ) : antidiagonal n ≃ Fin (n + 1) where
```

**Pull request closed without merging**